portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsWorkflow.testcase

```diff
@@ -96,11 +96,82 @@ definition {
 		}
 	}
 
+	@description = "LPS-147197. Verify rejected remote app portlet cannot be added to page"
+	@priority = "5"
+	test AdminCanRejectRemotePortlet {
+		task ("Given workflow approval process applied to remote apps") {
+			RemoteAppsWorkflow.applyWorkflow();
+		}
+
+		task ("And given a user with permissions to create remote app") {
+			Permissions.setUpRegRoleLoginUserCP(
+				roleTitle = "New Regular Role",
+				userEmailAddress = "userea@liferay.com",
+				userFirstName = "userfn",
+				userLastName = "userln",
+				userScreenName = "usersn");
+
+			RemoteAppsWorkflow.addUserPermissions();
+		}
+
+		task ("And given a remote app created as user with permissions") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
+
+			RemoteApps.goToRemoteAppsPortlet();
+
+			RemoteAppsWorkflow.addApp(
+				entryName = "Test Remote App",
+				entryURL = "http://www.liferay.com");
+		}
+
+		task ("When admin rejects remote app") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "Test Test");
+
+			Notifications.gotoNotifications();
+
+			AssertClick(
+				locator1 = "Notifications#NOTIFICATIONS_WORKFLOW_TITLE",
+				value1 = "userfn userln sent you a Client Extension Entry for review in the workflow.");
+
+			LexiconEntry.gotoEllipsisMenuItem(menuItem = "Assign to Me");
+
+			Workflow.confirmWorkflowAction();
+
+			LexiconEntry.gotoEllipsisMenuItem(menuItem = "Reject");
+
+			Workflow.confirmWorkflowAction();
```

Comment:
I feel like these steps are pretty common in Workflow that there's a good chance there's already a macro that exists for this. Could you double check? Go ahead and forward either after incorporating changes or no changes if not available.